aggregate_enclave_scripts/aggregator.py

```diff
@@ -0,0 +1,118 @@
+import sys
+import http.client
+from flask import Flask, request, Response
+import json
+import uuid
+import threading
+import abc
+import numpy as np
+
+app = Flask(__name__)
+query_list = []
+
+class RequestThread(threading.Thread):
+    def __init__(self, controller, enclaves_in_query, query_object, privacy_budget):
+        threading.Thread.__init__(self)
+        self.enclaves_in_query = enclaves_in_query
+        self.controller = controller
+        self.query_object = query_object
+        self.privacy_budget = privacy_budget
+
+    def run(self):
+        # for controller in controllers:
+        # r = requests.post('http://' + self.controller + "/request_query", data=str(self.query_object))
+        h1 = http.client.HTTPConnection(self.controller)
+        h1.request("POST", "/request_query", json.dumps({'query_type':str(self.query_object), 'privacy_budget':self.privacy_budget})) 
+        r1 = h1.getresponse()
+        # print(r.text)
+        # r = json.loads(r.text)
+        # for enclave in r:
+        #     if enclave['response'] != 'no':
+        #         self.enclaves_in_query[self.enclave] = r['Finished']
+
+class Query(abc.ABC):
+    """
+    ABC is an abstract base class to define an interface for all queries
+    """
+    @abc.abstractmethod
+    def run_query(self, data):
+        pass
+
+    @abc.abstractmethod
+    def generate_noise(self, data):
+        pass
+
+    @abc.abstractmethod
+    def __repr__(self):
+        pass
+
+class Sum(Query):
+    def __init__(self):
+        self.id = uuid.uuid4()
+
+    def run_query(self, data):
+        #self.amount_of_noise = 0
+        total = 0
+        for value in data:
+            total += int(value)
+            # try:
+            #     value = float(data[enclave])
+            # except:
+            #     continue
+            #self.amount_of_noise += 1
+        return total
+
+    def generate_noise(self, data, privacy_budget):
+        sensitivity = 1
```

Comment:
We'd have to, for each query object, have a custom defined way to do it.

Reply:
I think that part of the algorithm design is to define how the sensitivity is calculated for a particular query. I haven't seen anything, at least with this exact name, in the prior papers. If this has not been defined before, it would be a novel contribution. But I am not sure what exactly you use it for and what the correct calculation would be...